### ISSUE TITLE
False-positive "Match must be exhaustive"

### ISSUE THREAD

**Issue by vlad20012** [intellij-rust/intellij-rust#3851]
```rust
enum E {
    A, B
}

fn main() {
    match (E::A, true) {
        (E::A, true) => {}
        (E::A, false) => {}
        (E::B, true) => {}
        (E::B, false) => {}
    }
}
```

![image](https://user-images.githubusercontent.com/3221931/57925851-4e056900-78b2-11e9-969d-73c49d68864d.png)

**Comment by ortem** [comment]
Duplicates #3776
